### PR TITLE
Historic queries

### PR DESCRIPTION
**Pull request by ebuchman**
Opening for review

### CODE REVIEW

File: cmd/dump.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+var dumpCmd = &cobra.Command{
+	Run:   DumpDatabase,
+	Use:   "dump",
+	Short: "Dump a database",
+	Long:  `Dump all of the data for an underlying persistent database`,
+}
+
+func init() {
+	RootCmd.AddCommand(dumpCmd)
+	dumpCmd.Flags().StringVarP(&dbName, "name", "n", "merkleeyes", "Name db dir")
```
Comment (melekes) [inline cmd/dump.go]: `Name db dir`? so is it a name or a dir? maybe just `database name`?

File: iavl_test.go [hunk 2]
```diff
@@ -77,6 +77,176 @@ func P(n *IAVLNode) string {
...
+			text := fmt.Sprintf("%d) %s", i, actions[i].comment)
+			t.Error(text)
+		}
+	}
+}
+
+func TestOrder(t *testing.T) {
+	var tree *IAVLTree = NewIAVLTree(0, nil)
+	var up bool
+	up = tree.Set([]byte("aaaaaa"), []byte("aaaaaa"))
```
Comment (melekes) [inline iavl_test.go]: table tests?

```go
args := []struct{
  key []byte
  value []byte
}{
  {[]byte("aaaaaa"), []byte("aaaaaa")},
  {[]byte("bbbbbb"), []byte("bbbbbb")},
}
```

https://talks.golang.org/2012/10things.slide#3

File: iavl_tree_dump.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+	// Generic key, but still a node
+	return fmt.Sprintf("IAVLNode: [height: %d, key: %s%X, value: %s, hash: %X, leftHash: %X, rightHash: %X]",
+		node.height, prefix, formattedKey, formattedValue, node.hash, node.leftHash, node.rightHash)
+}
+
+// Dump everything in the database
+func (t *IAVLTree) Dump(mapping *KeyValueMapping) {
+	if t.roots == nil {
+		fmt.Printf("No tree loaded into memory\n")
```
Comment (melekes) [inline iavl_tree_dump.go]: Is this an error? Or just a warning?

**Comment by zramsay** [comment]
https://github.com/tendermint/iavl/pull/1

**Pull request closed without merging**